core/store/src/trie/trie_recording.rs

```diff
@@ -738,3 +786,98 @@ mod trie_recording_tests {
         test_trie_recording_consistency(true, true, true);
     }
 }
+
+#[cfg(test)]
+mod memtrie_batch_iteration_tests {
```

Comment:
not sure if there is a better place for this?

Reply:
If the code we are dealing with is generic and not specific to memtries, we can put the tests either here or in iterator.rs

Reply:
I made it generic like the tests in iterator.rs. Maybe we should move it there?